src/cow-react/common/pure/CurrencyInputPanel/CurrencyPreview.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react'
+import * as styledEl from './styled'
+import { CurrencySelectButton } from '@cow/modules/swap/pure/CurrencySelectButton'
+import { formatSmartAmount } from 'utils/format'
+import { FiatValue } from 'components/CurrencyInputPanel/FiatValue'
+import { Trans } from '@lingui/macro'
+import { PriceImpact } from 'hooks/usePriceImpact'
+import { CurrencyInfo } from '@cow/common/pure/CurrencyInputPanel/typings'
+
+interface BuiltItProps {
+  className: string
+}
+
+export interface CurrencyPreviewProps extends Partial<BuiltItProps> {
+  id: string
+  currencyInfo: CurrencyInfo
+  priceImpactParams?: PriceImpact
+  topLabel?: string
+}
+
+export function CurrencyPreview(props: CurrencyPreviewProps) {
+  const { id, currencyInfo, className, priceImpactParams, topLabel } = props
+  const { priceImpact, loading: priceImpactLoading } = priceImpactParams || {}
+  const { currency, balance, fiatAmount, viewAmount, rawAmount } = currencyInfo
+
+  return (
+    <>
+      <styledEl.Wrapper id={id} className={className} withReceiveAmountInfo={false}>
+        {topLabel && <styledEl.CurrencyTopLabel>{topLabel}</styledEl.CurrencyTopLabel>}
+
+        <styledEl.CurrencyInputBox flexibleWidth={true}>
+          <div>
+            <CurrencySelectButton currency={currency || undefined} loading={false} readonlyMode={true} />
+          </div>
+          <div>
+            <styledEl.NumericalInput
+              title={rawAmount?.toExact() + ' ' + currency?.symbol}
```

Comment:
There's a util right that was created to deal with Fraction instances.
Not sure it'll fit the type, though https://github.com/cowprotocol/cowswap/blob/6f5e9a33f87ebb93fbbeb73dcbdd60e0f06376e0/src/custom/utils/format.ts#L171

Reply:
Here we should display an exact amount, so I think it's correct